Dikubot/Webapp/Data/FacebookService.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json.Serialization;
+using Dikubot.DataLayer.Database.Global.Facebook;
+using Microsoft.AspNetCore.Components;
+
+namespace Dikubot.Webapp.Data;
+
+public class FacebookService
+{
+
+    private string appId = Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID");
+    private string appSecret = Environment.GetEnvironmentVariable("FACEBOOK_APP_SECRET"); 
+    private string graphApiVersion = "v16.0";
+    
+    private FacebookPageService _facebookPageService { get; set; }
+    private HttpClient _httpClient { get; set; }
+    private NavigationManager _navigationManager { get; set; }
+
+    public FacebookService(HttpClient httpClient, FacebookPageService facebookPageService, NavigationManager navigationManager)
+    {
+        _httpClient = httpClient;
+        _facebookPageService = facebookPageService;
+        _navigationManager = navigationManager;
+    }
+    
+    private Dictionary<string, List<FBEvent>> FBEventDic = new Dictionary<string, List<FBEvent>>();
+
+    public List<FBEvent> GetFBEvents(string id)
+    {
+        
+        if (FBEventDic.ContainsKey(id))
+        {
+            return FBEventDic[id];
+        }
+        var accessToken = _facebookPageService.Get(m => m.PageId == id).AccessToken;
+        
+        var requestUri = $"https://graph.facebook.com/{graphApiVersion}/{id}/events?access_token={accessToken}";
+        var result = _httpClient.GetFromJsonAsync<FBRespons<FBEvent>>(requestUri).Result.Data;
+        FBEventDic.Add(id, result);
+        return result;
+    }
+    
+    public string GetFBPicture(string id)
+    {
+        
+        var accessToken = _facebookPageService.Get(m => m.PageId == id)?.AccessToken;
+        
+        if (!string.IsNullOrWhiteSpace(accessToken))
+        {
+            var link = $"https://graph.facebook.com/{graphApiVersion}/{id}/picture?redirect=0&access_token={accessToken}";
+            return _httpClient.GetFromJsonAsync<FBPictureRespons>(link).Result.Data.Url;
+            
+        }
+        return "";
+    }
+
+    public void ConnectFacebookAction(string redirectUri)
+    {
+        string link =
+            $"https://www.facebook.com/{graphApiVersion}/dialog/oauth?" +
+            "client_id=" + Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID") +
```

Comment:
Move Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID")  out of here. It is a constant.